client/sysinfo/src/sysinfo.rs

```diff
@@ -337,20 +337,23 @@ pub fn benchmark_disk_random_writes(directory: &Path) -> Result<u64, String> {
 		.map(|s| s as u64)
 }
 
-/// Verify signatures of different random data `reps` many times.
 pub fn benchmark_sr25519_verify(
-	reps: usize,
-	duration: Duration,
-	input_len: usize,
+	max_iterations: Option<usize>,
+	max_duration: Option<Duration>,
 ) -> Result<f64, String> {
+	if max_iterations.is_none() && max_duration.is_none() {
+		return Err("max_iterations and max_duration cannot both be None".into())
+	}
```

Comment:
Hmm.... I totally get why you did this, as if both are `None` then the benchmark will run forever. It would be nice to get rid of this `Result` though.... maybe we could use an enum instead? Something like this perhaps:

```rust
enum ExecutionLimit {
    ByMaxDuration(Duration),
    ByMaxIterations(usize),
    ByBoth {
        max_iterations: usize,
        max_duration: Duration
    }
}
```

Reply:
Nice idea. Yea will do.